packages/RNTester/e2e/__tests__/Modal-test.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails oncall+react_native
+ * @format
+ */
+
+/* global device, element, by, expect */
+const {
+  openComponentWithLabel,
+  openExampleWithTitle,
+} = require('../e2e-helpers');
+
+describe('Modal', () => {
+  beforeAll(async () => {
+    await device.reloadReactNative();
+    await openComponentWithLabel(
+      '<Modal>',
+      '<Modal> Component for presenting modal views.',
+    );
+  });
+
+  it('should show content on button click', async () => {
+    await openExampleWithTitle('Basic Modal');
+    await element(by.text('READ MORE')).tap();
+    await expect(element(by.text('PROCEED'))).toBeVisible();
+    await element(by.text('PROCEED')).tap();
+  });
+
+  it('should cover full screen with statusBarTranslucent enabled', async () => {
+    await openExampleWithTitle('Status Bar Behavior with Modal');
+    await element(by.text('OPEN IN FULL SCREEN VIEW')).tap();
+    await expect(element(by.text('PROCEED'))).toBeVisible();
+    await element(by.text('PROCEED')).tap();
+  });
+
+  it('should listen and launch action for back button press', async () => {
+    await openExampleWithTitle('Actions on physical back button');
+    await element(by.text('LISTEN FOR CLOSE ACTION')).tap();
+    await device.pressBack();
+    await element(by.text('OK')).tap();
+    await expect(element(by.text('PROCEED'))).toBeVisible();
+    await element(by.text('PROCEED')).tap();
+  });
+
+  it('should listen for onShow event', async () => {
+    await openExampleWithTitle('Modal OnShow Actions');
+    await element(by.text('OPEN MODAL')).tap();
+    await expect(element(by.text('Welcome User'))).toBeVisible();
+    await element(by.text('OK')).tap();
+    await expect(element(by.text('PROCEED'))).toBeVisible();
+    await element(by.text('PROCEED')).tap();
+  });
+});
```

Comment:
@YashKumarVerma I think we can have a test to ensure that the modal closes too. Using expect().not.toBeVisible() !

Reply:
nice idea, i'd be doing that.